### DISCUSSION TITLE
Nuxt 3 site with nomodule/legacy support

### DISCUSSION THREAD

**Discussion by dombarnes** [nuxt/nuxt#20269] [Questions]
I've recently migrated a vanilla Vue 3/Webpack project to Nuxt 3/Vite and am having issues with running my site on a specifically old browser. While it supports many features of ES6, it does NOT support module declarations for JS files. As such, it won't load my site.

I can't seem to figure out any way to get my project to produce the assets I used to get. My old build would generate 
```
<script defer="defer" type="module" src="/js/chunk-vendors.js"></script>
<script defer="defer" type="module" src="/js/app.js"></script>
<script defer="defer" src="/js/chunk-vendors-legacy.js" nomodule></script>
<script defer="defer" src="/js/app-legacy.js" nomodule></script>
```
I did try @vite/legacy-module but that didn't seem to do what I think it should. It produces
Config
```
vite: {
    plugins: [
      legacy({
        targets: ['defaults']
      })
    ]
  }
```
```
<script type="module" src="/_nuxt/entry-legacy.b8bc2dd9.js" crossorigin></script>
<script type="module" src="/_nuxt/entry.40d09d1a.js" crossorigin></script>
<script type="module" src="/_nuxt/polyfills-legacy.182f7f7b.js" crossorigin></script></body>
```
Maybe its not possible, maybe its a but in @vite/plugin-legacy, or maybe I'm just totally missing the mark somewhere on how all the pieces work together.

**Comment by jackie-rachel** [comment]
I have a same question.do you work it out now?

**Reply by dombarnes** [reply]
I ended up adding the module [detailed here](https://github.com/nuxt/nuxt/issues/15464#issuecomment-1490325832) and in my `nuxt.config.js` (not sure which bits solved it but this seems to output what I need)
```js
import legacy from '@vitejs/plugin-legacy'
export default defineNuxtConfig({
... (sections omitted)
modules: [
    '@nuxtjs/tailwindcss',
    '~/modules/fixViteLegacyPlugin.ts'
  ],
...
 nitro: {
    preset: 'azure',
    esbuild: {
      options: {
        target: 'es5'
      }
    }
  },
...
vite: {
    build: {
      // target: ['esnext', 'ios12']
    },
    optimizeDeps: {
      exclude: ['fsevents']
    },
    plugins: [
      legacy({
        targets: ['chrome >= 64', 'edge >= 79', 'safari >= 11.1', 'firefox >= 67'],
        ignoreBrowserslistConfig: true,
        renderLegacyChunks: true,
        modernPolyfills: ['es/global-this']
      })
    ]
  }
